Added the 'bare' modifier to CLI 'diff' / 'dump' commands.

mikeller opened the pull request.

etracer65 commented:
There seems to be some inconsistency with the `batch start` for the `dump hardware` (and `diff hardware`). In current master there are no `batch` statements in the output. But in this branch there are `batch start` commands included but no `batch end` or `save` in the output. I'm not sure if it should have batch statements, but if so it should be consistent.

And argument for having batch statements is that I actually found an error while testing the unified targets creation documentation. I made a `REVO` target from this branch and flashed it to a board. Then did the `dump hardware` and saved it (this include the `batch start` command). Then built a `STM32F405` target and flashed it to the board. Then went back in and tried to apply the saved `dump hardware`. Got the following error:
```
# resource ESCSERIAL 1 B14
###ERROR: INVALID RESOURCE NAME: 'ESCSERIAL'###
```
The `batch start` actually helped me find this even though it scrolled off the screen because when I tried to save I got:
```
###ERROR: ERRORS WERE DETECTED - PLEASE REVIEW BEFORE CONTINUING###
```
So two things: the inconsistent use of the `batch` commands in this PR, and not related to this PR the `ESCSERIAL` resource error.

mikeller commented:
Yes, the bug affecting `batch stop` is known, and fixed in #7742. I think error checking should be enabled for `dump hardware`. There will be errors like the one you found, caused by targets using features not yet supported on Unified Targets, but it is better if target maintainers know about these.

mikeller commented:
@etracer65: Try now.

etracer65 commented:
If the logic is getting too complex with all the `dump`/`diff` options, maybe we should concentrate it on whether the `batch start` should be included and if so set a flag that drives adding the `batch end` to the bottom rather than having to reevaluate all the `dump`/`diff` options a second time at the end. That way we ensure there's always a matching end if the `batch start` was output.

etracer65 commented:
Also want to once again suggest we consider reviving https://github.com/betaflight/betaflight-configurator/pull/1193 as batch command processing was originally intended. This will take care of automatically wrapping any pasted blocks of commands with the appropriate `batch` commands.

This would then negate the need to add the `batch` commands to the `diff`/`dump` output and avoid any complications it causes.

mikeller commented:
@etracer65:

> Also want to once again suggest we consider reviving betaflight/betaflight-configurator#1193 as batch command processing was originally intended. This will take care of automatically wrapping any pasted blocks of commands with the appropriate batch commands.

> This would then negate the need to add the batch commands to the diff/dump output and avoid any complications it causes.

If we went for what I always suggested, and set error checking on `save` to be always on, and added a proper command to force override errorneous input, we would also not have to inject commands into the CLI output, *and* we would not have to make configurator inject unrequested commands without the user's knowledge, making CLI behave erratically from the user's perspective.

Anyhow, I've changed it as requested, and also reordered the start of the output, so that `board_name`, `manufacturer_id`, `mcu_id`, and `signature` are now covered as well, which is their intended purpose.

mikeller commented:
@etracer65: Rebased.
